apis/config.go

```diff
@@ -288,3 +305,19 @@ func defaultExternalAddress() (net.IP, error) {
 	}
 	return ip, nil
 }
+
+func defaultListenAddress(cfg *EtcdAdmConfig) (net.IP, error) {
+	if cfg.ListenAddress == "" {
+		return defaultExternalAddress()
+	} else {
+		return netutil.ChooseBindAddress(net.ParseIP(cfg.ListenAddress))
```

Comment:
`--listen-address 127.0.0.1` is a no-op, because of `ChooseBindAddress`. I think users should be allowed to use `127.0.0.1` as a listen address.

What about implementing `defaultExternalAddress` as in [kube-apiserver](https://github.com/kubernetes/kubernetes/blob/a66c6e459cad47ce39eb3a6d3f6db380ac1db21c/staging/src/k8s.io/apiserver/pkg/server/options/serving.go#L111)?

Then `defaultListenAddress` and `defaultAdvertiseAddress` can just wrap that and add nice context to errors.